packages/fx-core/src/plugins/resource/apiconnector/sampleHandler.ts

```diff
@@ -51,22 +52,19 @@ export class SampleHandler {
         component: this.component,
         languageExt: this.languageExt,
       };
-      const codeFileName: string = config.APIName + "." + fileSuffix;
-      let codePath = path.join(this.projectRoot, this.component);
-      if (await fs.pathExists(path.join(codePath, "src"))) {
-        codePath = path.join(codePath, "src");
+      const sampleFileName: string = config.APIName + "." + fileSuffix;
+      const componentPath = path.join(this.projectRoot, this.component);
+      const sampleDirPath = getSampleDirPath(componentPath);
+      fs.ensureDir(sampleDirPath);
+      const sampleFilePath = path.join(sampleDirPath, sampleFileName);
+      if (await fs.pathExists(sampleFilePath)) {
```

Comment:
should we throw error instead of delete existing file? otherwise we may delete user's file

Reply:
We already check the duplicate files in question stage. 
Here is an inner logic to ensure the writing execution will succeed.

Reply:
can we throw error here in case the question stage didn't find duplicate files correctly?

Reply:
Will update it in a new PR.